R/write_df_from_fastq.R

```diff
@@ -113,3 +113,99 @@ write_df_from_fastq <- function(
   
   return(cumulative_count_df)
 }
+
+
+# this function is for DRAGEN-formatted files
+write_df_from_fastq_DRAGEN <- function(
+    forward_read_fastq_files,
+    write_interval = NA,
+    CL_BC_LENGTH = 24,
+    PLATE_BC_LENGTH = 8,
+    WELL_BC_LENGTH = 8,
+    save_loc = getwd()
+) 
+{
+  write_interval = as.numeric(write_interval)
+  
+  n_total_reads <- 0
+  cumulative_count_df <- list()
+  
+  lp = 1
+  for (i in 1:length(forward_read_fastq_files)) {
+    forward_stream <- ShortRead::FastqStreamer(forward_read_fastq_files[i])
+    
+    print(paste0("forward read file is: ", forward_read_fastq_files[i]))
+    
+    j <- 0
+    repeat{
+      forward_reads_chunk <- ShortRead::yield(forward_stream)
+      
+      if (length(forward_reads_chunk) == 0) {break}
+      
+      print(paste0('Processing read ', j * 1e6, ' to ', 
+                   j*1e6 + length(forward_reads_chunk), 
+                   ' from file ', i)) #(ShortRead)The default size for both streams and samples is 1M records;
+      j <- j + 1
+      
+      forward_reads_string_set <- forward_reads_chunk  %>%
+        ShortRead::sread() %>%
+        Biostrings::DNAStringSet()
+      
+      forward_reads_cl_barcode <- XVector::subseq(forward_reads_string_set, 1, CL_BC_LENGTH)
+      
+      forward_reads_cl_barcode = as.character(forward_reads_cl_barcode)
+      
+      # get the index 1 and index 2 barcodes from the fastq file
+      print("Get index 1 and index 2 barcodes from fastq file")
+      
+      # do the below string splitting but for each read in the fastq file
+      temp <- as.character(forward_reads_chunk@id)
+      
+      fastq_header_split <- function(str){
```

Comment:
No need for the Sample Sheet then? Is this assuming that the file name contains the Index1 and Index2 then

Reply:
If so, might just want to add that as a comment, otherwise this looks good to me